module/users/helper.js

```diff
@@ -856,7 +856,7 @@ module.exports = class UsersHelper {
         userRelatedPrograms = userRelatedPrograms.slice(startIndex,endIndex) 
         
         let userRelatedProgramsData = await programsHelper.programDocuments(
-          { _id: { $in: userRelatedPrograms } },
+          { _id: { $in: userRelatedPrograms }, isAPrivateProgram: false },
```

Comment:
@ankitpws it has to be put in the query for non-targeted programs likely getUserJoinedProgramDetailsWithPreviousProfiles not here

Reply:
Added in getUserJoinedProgramDetailsWithPreviousProfiles